apps/emqx_coap/src/emqx_coap_resource.erl

```diff
@@ -103,13 +103,20 @@ coap_unobserve({state, ChId, Prefix, Topic}) ->
     ?LOG(error, "ignore unknown unobserve request ChId=~p, Prefix=~p, Topic=~p", [ChId, Prefix, Topic]),
     ok.
 
-handle_info({dispatch, Topic, Payload}, State) ->
-    ?LOG(debug, "dispatch Topic=~p, Payload=~p", [Topic, Payload]),
-    {notify, [], #coap_content{format = <<"application/octet-stream">>, payload = Payload}, State};
+handle_info({dispatch, Msg}, State) ->
+    Payload = emqx_coap_mqtt_adapter:message_payload(Msg),
```

Comment:
should we keep the debug log

Reply:
There's already a similar debug log in the `do_deliver` function of the `emqx_coap_mqtt_adapter` module:
```
    ?LOG(debug, "deliver message from broker, msg: ~p", [Msg]),
```